cmd/auth/main.go

```diff
@@ -42,6 +42,9 @@ func main() {
 	authContextResolverFactory := resolver.AuthContextResolverFactory{}
 	srv := server.NewServer("localhost:3000", authContextResolverFactory)
 
+	// RecoverMiddleware must come first
+	srv.Use(middleware.RecoverMiddleware{}.Handle)
```

Comment:
Seem we will not have case that don't want a `Recovery`.

I am thinking instead of leaving a comment to `RecoverMiddleware must come first`. Will we put it as a parameter of `NewServer` as `authContextResolverFactory`? It seem easier to enforce the existence and order.